### PR TITLE
mgr/crash: Calculate and add stack_sig to metadata

### PR DESCRIPTION
**Pull request by dmick**
Signed-off-by: Dan Mick <dan.mick@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/crash/module.py [hunk 1]
```diff
@@ -149,6 +181,8 @@ def do_post(self, cmd, inbuf):
             metadata = self.validate_crash_metadata(inbuf)
         except Exception as e:
             return errno.EINVAL, '', 'malformed crash metadata: %s' % e
+        metadata['stack_sig'] = \
+            self.calc_sig(metadata['backtrace'], metadata['assert_msg'])
```
Comment (liewegas) [inline src/pybind/mgr/crash/module.py]: assert_msg not always present..
```
2019-11-05T23:49:12.433 INFO:tasks.ceph.mgr.x.smithi143.stderr:2019-11-05T23:49:12.442+0000 7f56ec6fe700 -1 mgr.server reply reply (22) Invalid argument Traceback (most recent call last):
2019-11-05T23:49:12.434 INFO:tasks.ceph.mgr.x.smithi143.stderr:  File "/usr/share/ceph/mgr/mgr_module.py", line 975, in _handle_command
2019-11-05T23:49:12.434 INFO:tasks.ceph.mgr.x.smithi143.stderr:    return self.handle_command(inbuf, cmd)
2019-11-05T23:49:12.434 INFO:tasks.ceph.mgr.x.smithi143.stderr:  File "/usr/share/ceph/mgr/crash/module.py", line 107, in handle_command
2019-11-05T23:49:12.434 INFO:tasks.ceph.mgr.x.smithi143.stderr:    return handler(self, command, inbuf)
2019-11-05T23:49:12.434 INFO:tasks.ceph.mgr.x.smithi143.stderr:  File "/usr/share/ceph/mgr/crash/module.py", line 185, in do_post
2019-11-05T23:49:12.434 INFO:tasks.ceph.mgr.x.smithi143.stderr:    self.calc_sig(metadata['backtrace'], metadata['assert_msg'])
2019-11-05T23:49:12.434 INFO:tasks.ceph.mgr.x.smithi143.stderr:KeyError: ('assert_msg',)
```

Comment (dmick) [reply src/pybind/mgr/crash/module.py]: Oops, will fix

Comment (dmick) [reply src/pybind/mgr/crash/module.py]: Pushed new commit that should address this.

**Comment by dmick** [comment]
This should be OK now I think.  I can squash and repush if you like.

**Pull request merged**